semantics/psc-trees-annotation.adb

```diff
@@ -110,6 +110,8 @@ package body PSC.Trees.Annotation is
          Lists.Replace_Nth_Element (T.Annotations, N, New_Operand);
    end Set_Nth_Operand;
 
+   function Kind (T : Tree) return Tree_Kind_Enum is (Annotation_Kind);
```

Comment:
I suggest you move these expression functions to the spec.  There is no need for a separate body for these functions, since they are not doing anything needing information hiding.